fidesctl/src/fidesctl/cli/__init__.py

```diff
@@ -31,26 +31,42 @@
     default="",
     help="Optional configuration file",
 )
+@click.option(
+    "--local",
+    is_flag=True,
+    help="Runs in a local mode that doesn't utilize server calls.",
+)
 @click.pass_context
-def cli(ctx: click.Context, config_path: str) -> None:
+def cli(ctx: click.Context, config_path: str, local: bool) -> None:
     """
-    The parent group for the Fides CLI.
-    Loads the config and passes it within the context.
+    The parent group for the Fidesctl CLI.
     """
+
+    local_commands = [evaluate, parse, view_config]
+    all_commands = [
+        annotate_dataset,
+        apply,
+        delete,
+        generate_dataset,
+        get,
+        init_db,
+        ls,
+        ping,
+        reset_db,
+        webserver,
+    ] + local_commands
+
     ctx.ensure_object(dict)
-    ctx.obj["CONFIG"] = get_config(config_path)
-
-
-cli.add_command(apply)
-cli.add_command(delete)
-cli.add_command(evaluate)
-cli.add_command(generate_dataset)
-cli.add_command(annotate_dataset)
-cli.add_command(get)
-cli.add_command(init_db)
-cli.add_command(ls)
-cli.add_command(parse)
-cli.add_command(ping)
-cli.add_command(reset_db)
-cli.add_command(view_config)
-cli.add_command(webserver)
+    config = get_config(config_path)
+
+    if local or config.cli.local_mode:
+        config.cli.local_mode = True
+        for command in local_commands:
+            cli.add_command(command)
+    else:
+        for command in all_commands:
+            cli.add_command(command)
```

Comment:
```suggestion
    for command in local_commands:
    	config.cli.local_mode = True
    	cli.add_command(command)
    	
    if not (local or config.cli.local_mode):
    	config.cli.local_mode = False
    	for command in api_commands:
    		cli.add_command(command)
```
If my above suggestion to completely separate the commands by "localness" is accepted, I think this is a more intuitive way to code this out. To me, your version reads as "we either want only the local commands or all commands", but this version reads like "yea, we always want the local commands, but we _might_ also want the API commands". I also like that we explicitly set `config.cli.local_mode` in both cases.

Reply:
this is exactly what I was trying to do but couldn't think of the way to do it, thank you!